### PR TITLE
refactor(simple auth): API v2 adapter

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

**Review by 1yefuwang1 — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/simpleauth/v2/index.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  ResourcePluginsV2,
+} from "../../../solution/fx-solution/ResourcePluginContainer";
+import {
+  configureLocalResourceAdapter,
+  generateResourceTemplateAdapter,
+  provisionLocalResourceAdapter,
+} from "../../utils4v2";
+
+@Service(ResourcePluginsV2.AadPlugin)
+export class AadPluginV2 implements ResourcePlugin {
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/simpleauth/v2/index.ts:31]: why aad?

File: packages/fx-core/src/plugins/resource/utils4v2.ts [hunk 2]
```diff
@@ -147,20 +142,80 @@ export async function deployAdapter(
...
+    teamsApp: ConfigMap.fromJSON(localSettings.teamsApp)!,
+    auth: ConfigMap.fromJSON(localSettings.auth),
+    backend: ConfigMap.fromJSON(localSettings.backend),
+    bot: ConfigMap.fromJSON(localSettings.bot),
+    frontend: ConfigMap.fromJSON(localSettings.frontend),
+  };
+  pluginContext.appStudioToken = tokenProvider.appStudioToken;
+  pluginContext.azureAccountProvider = tokenProvider.azureAccountProvider;
+  pluginContext.graphTokenProvider = tokenProvider.graphTokenProvider;
+  const res = await plugin.localDebug(pluginContext);
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/utils4v2.ts:164]: where is localDebug defined?

File: packages/fx-core/src/plugins/resource/simpleauth/v2/index.ts (hunk 1 above)
Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/simpleauth/v2/index.ts:31]: It's a bug

Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/simpleauth/v2/index.ts:31]: I will fix it

File: packages/fx-core/src/plugins/resource/utils4v2.ts (hunk 2 above)
Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/utils4v2.ts:164]: provisionLocalResources is replacement of localDebug

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/utils4v2.ts:164]: I mean plugin.localDebug here. I can not find the implementation.

**Review by KennethBWSong — APPROVED** [review]

**Pull request merged**